nucypher/network/nodes.py

```diff
@@ -1012,9 +1013,19 @@ def _staker_is_really_staking(self, registry: BaseContractRegistry) -> bool:
         As a follow-up, this checks that the staker is, indeed, staking.
         """
         # Lazy agent get or create
-        staking_agent = ContractAgency.get_agent(StakingEscrowAgent, registry=registry)
-        locked_tokens = staking_agent.get_locked_tokens(staker_address=self.checksum_address)
-        return locked_tokens > 0  # TODO: Consider min stake size #1115
+        staking_agent = ContractAgency.get_agent(StakingEscrowAgent, registry=registry)  # type: StakingEscrowAgent
+
+        try:
+            economics = TokenEconomicsFactory.get_economics(registry=registry)
+        except Exception:
+            raise  # TODO: Get StandardEconomics
+
+        min_stake = economics.minimum_allowed_locked
+
+        stake_current_period = staking_agent.get_locked_tokens(staker_address=self.checksum_address, periods=0)
+        stake_next_period = staking_agent.get_locked_tokens(staker_address=self.checksum_address, periods=1)
+        is_staking = max(stake_current_period, stake_next_period) >= min_stake
```

Comment:
Just confirming what we are checking for here - Is our goal to make sure that the stake for this period and the next are both greater than `min_stake`? If so, should we instead use min? i.e.:
```suggestion
        is_staking = min(stake_current_period, stake_next_period) >= min_stake
```

Or is it that the stake for the current period could be 0 (not staking until UTC midnight), and the stake for the next period is correctly set, in which case `max` will usually just produce the stake for the next period?

Reply:
It's the latter, yes.